onnxoptimizer/passes/eliminate_common_subexpression.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+#pragma once
+
+#include <unordered_map>
+
+#include "onnx/defs/tensor_util.h"
+#include "onnxoptimizer/pass.h"
+#include "onnxoptimizer/passes/logging.h"
+#include "onnxoptimizer/passes/node_hash.h"
+#include "onnxoptimizer/passes/pass_util.h"
+#include "onnxoptimizer/passes/string_utils.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct EliminateCommonSubexpression final : public FullGraphBasedPass {
+  explicit EliminateCommonSubexpression()
+      : FullGraphBasedPass(PassType::Nop, PassEfficiency::Complete,
+                           PassOptimizationType::Compute) {}
+  std::string getPassName() const override {
+    return "eliminate_common_subexpression";
+  }
+  PassAnalysisType getPassAnalysisType() const override {
+    return PassAnalysisType::CountBased;
+  }
+
+  unsigned int EliminateCSE(Graph &graph) {
+    auto node_list = graph.nodes();
+    unsigned int cse_removed = 0;
+    std::unordered_map<size_t, Node *> hash_map;
```

Comment:
这里需要改一下，应该直接用
```suggestion
    std::unordered_map<Node, Node *, CSEHash, CSEEqual> map;
```
hash 只是加速查找的实现细节，本质是从 map 里找到一个和当前 node 等价（`operator==`）的 node。另外因为这个 PR 实现 hash 和判等都是面向 cse 这个具体业务的，所以不适合作为 Node 类的标准 hash 和 operator==，作为普通函数、unordered_map 的参数更合适